spec/datadog/di/transport_spec.rb

```diff
@@ -0,0 +1,96 @@
+require "datadog/di/spec_helper"
+require "datadog/di/transport"
+
+RSpec.describe Datadog::DI::Transport do
+  let(:agent_settings) do
+    double('agent settings')
+  end
+
+  describe '.new' do
+    it 'creates an instance using agent settings' do
+      expect(agent_settings).to receive(:hostname).and_return('localhost')
+      expect(agent_settings).to receive(:port).and_return(8126)
+      expect(agent_settings).to receive(:timeout_seconds).and_return(1)
+      expect(agent_settings).to receive(:ssl).and_return(false)
+
+      expect(described_class.new(agent_settings)).to be_a(described_class)
+    end
+  end
```

Comment:
Could we avoid this test because it's not testing much.

Reply:
This is actually testing that the agent settings are used for the transport.